packages/engine/src/scene/systems/TriggerSystem.ts

```diff
@@ -1,41 +1,81 @@
 import { Entity } from '../../ecs/classes/Entity'
```

Comment:
the changes to this system must be reverted if this branch is to be merged, the existing functionality of triggers should cover all use cases. Please re-think the implementation here.